### ISSUE TITLE
object-curly-spacing+arraysInObjects thinks `{ a: list[0] }` delimiter is an array.

### ISSUE THREAD

**Issue by Droogans** [eslint/eslint#5538]
**What version of ESLint are you using?**

v2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

I'm using ESLint to parse my source code.

**What did you do? Please include the actual source code causing the issue.**

When evaluating

```js
"object-curly-spacing": [2, "always", {
    "arraysInObjects": false
}]
```

I am being told that my array index operation in an inline object is an "arrayInObject".

```js
var a = { thingInList: list[0] };
```

**What did you expect to happen?**

I expected ESLint to understand the difference between a closing bracket in an array index access operation and a javascript array.

**What actually happened? Please include the actual, raw output from ESLint.**

It assumed the array index access brackets were part of a javascript array.

**Comment by BYK** [comment]
I was able to reproduce this on latest master with the following new test case under `valid`:

```js
{ code: "var a = { thingInList: list[0] };", options: ["always", {"arraysInObjects": false}] },
```

**Comment by Droogans** [comment]
@BYK Thanks! That was quick.
